### DISCUSSION TITLE
Custom Ports in Roadwarrior config

### DISCUSSION THREAD

**Discussion by walkim14** [strongswan/strongswan#1889] [Q&A] [answered]
Hello lovely Strongswan community!
I've been cracking my head the last few days trying to get strongswan running with custom ports, since I want to get around university firewall restrictions. I know this should be possbile with the default socket implementation and it works without trying any port changing, but as soon as I try and change the ports everything stops working and I have tried a hundred different combinations and read every forum thread (somehow Tobias Brunner is in all of them). I need your help Strongsswan gods.
In the file I attached below is my server and client config aswell as the logs. If you need anyhting else please just let me know.
Thank you in advance.
[strongswan support.txt](https://github.com/strongswan/strongswan/files/12546938/strongswan.support.txt)

**Comment by tobiasbrunner** [comment, marked as answer]
Your ipsec.conf files look OK (there are some things that don't make sense, like `rightdns` on the client, or `rightfirewall` on either, and `leftikeport` on the server is also not necessary).

The port config is also fine (although, it's not necessary to use the same ports on client and server). However, make sure the daemon on the server is actually listening on port 3500 (you can do so e.g. via `sudo ss -ulnp`). Depending on the contents of `/etc/strongswan.conf`, the `charon.conf` snippet might not be included and applied, so make sure it is.

Then try to verify whether packets from the client actually arrive at the server (e.g. via `tcpdump`). If not, there is a firewall that blocks these packets for some reason.

**Reply by walkim14** [reply]
Thanks for pointing out my useless config options, in my desperation the last few days I just started throwing shit at the wall at some point.
The output from `ss -ulnp`  shows that charon is listening on ports 3500 and 3501. But when looking at the traffic with either tcpdump or wireshark no packets are arriving at the server when trying to initialize a connection with the client. I am not that well versed in networking, but I tried a multitude of ports until now and none of them seem to get through. The standart 500 and 4500 ports work beautifully, but anything else seems to get stuck. Do have any suggestions on what to try next? My iptable rules on both client and server allow anything and everthing to go through.
Is there maybe a way to see where the packets get stuck?

**Reply by walkim14** [reply]
Also here is my strongswan.conf (I didn't modify anything here)
`charon {
        load_modular = yes
        plugins {
                include strongswan.d/charon/*.conf
        }
}
include strongswan.d/*.conf`

**Reply by tobiasbrunner** [reply]
> The standart 500 and 4500 ports work beautifully, but anything else seems to get stuck.

Hm, so why do you want to use custom ports? Or are you testing in a different environment?

> Do have any suggestions on what to try next? My iptable rules on both client and server allow anything and everthing to go through.

This indicates that some firewall between your client and server blocks UDP except for some specific and well-known ports. You could try some other port that might be open (e.g. DNS/53, or one that is usually only used with TCP but maybe is open for UDP, like SMTP/25, HTTP/80 etc.).

> Is there maybe a way to see where the packets get stuck?

Not really, unless you have access to a router/firewall between your client and server to check what's going on there.

**Reply by walkim14** [reply]
I am working on something that will be deployed in multiple networks around the world (including some universities) and from experience we know that especially unis can be very restrictive with what ports are open in their network and are hard to work with when saying that a port has to be open for our product to work.
Thank you for quick response, I will keep trying, now that I know that it should be working from the strongswan config I have, I shoud be able to resolve this. 
Again thank you for helping me out :)
